semantic_release/cli/commands/changelog.py

```diff
@@ -62,10 +52,55 @@ def changelog(ctx: click.Context, release_tag: str | None = None) -> None:
             )
             ctx.exit(0)
 
-        changelog_text = render_default_changelog_file(env)
-        with open(str(changelog_file), "w+", encoding="utf-8") as f:
-            f.write(changelog_text)
+        with open(str(changelog_file), "a+", encoding="utf-8") as f:
+            f.seek(0)
+            existed_changelog_text = f.read()
+            f.truncate()
```

Comment:
This is error-prone - if there's an exception later on then we will have truncated the changelog without writing any text back to it